app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -215,7 +247,8 @@ private void addNewDevice(final Intent intent) {
         final BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
 
         // Skip paired devices and devices without a name.
-        if (device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
+        if (device != null && device.getName() != null
+            && device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
```

Comment:
Could you maybe refactor this into a separate method named `isDevicePaired` or something like that, this if-statement is getting a bit hard to read 😅